code/modules/atmospherics/portable/portable_pressurizer.dm

```diff
@@ -0,0 +1,429 @@
+					air_contents.merge(removed)
+
+	proc/is_air_safe()
+		var/total_moles = max(TOTAL_MOLES(src.air_contents),1)
+		return(((src.air_contents.toxins/total_moles) < 0.01) && ((src.air_contents.carbon_dioxide/total_moles) < 0.05) && (src.air_contents.temperature < FIRE_MINIMUM_TEMPERATURE_TO_SPREAD))
+
+	proc/process_raw_materials()
+		if(status & NOPOWER)
+			return
+
+		if(!target_material)
+			material_progress = 0
+			if(length(src.contents))
+				target_material = pick(src.contents)
+				if((target_material.amount > 1) && (target_material.material?.name in src.whitelist))
+					var/atom/movable/splitStack = target_material.split_stack(target_material.amount-1)
+					splitStack.set_loc(src)
+				target_material.set_loc(null)
+			else
+				process_materials = PROCESS_OFF
+				return
+
+		if(MIXTURE_PRESSURE(air_contents) > maximum_pressure*2)
+			process_materials = PROCESS_PAUSED
+			return
+		else if(material_progress < 100)
+			process_materials = PROCESS_ACTIVE
+			var/progress = min(src.process_rate * 5,100-material_progress)
+			var/datum/gas_mixture/GM = unpool(/datum/gas_mixture)
+			GM.temperature = T20C
+			if(target_material.material?.name in src.whitelist)
+				switch(target_material.material.name)
+					if("molitz")
+						GM.oxygen += 1500 * progress / 100
+					if("viscerite")
+						GM.nitrogen += 1500 * progress / 100
+					if("char")
+						GM.carbon_dioxide += 500 * progress / 100
+					if("plasmastone")
+						GM.toxins += 500 * progress / 100
+			else
+				playsound(src.loc, 'sound/machines/buzz-two.ogg', 20)
+				process_materials = PROCESS_PAUSED
+				target_material?.set_loc(src.loc)
+				target_material = null
+				return
+			src.air_contents.merge(GM)
+			material_progress = clamp(material_progress + progress, 0, 100)
+			use_power(src.powconsumption)
+		else
+			qdel(target_material)
+			target_material = null
+
+	emag_act(var/mob/user, var/obj/item/card/emag/E)
+		if (!src.emagged)
+			if (user)
+				user.show_text("You short out the material processor on [src].", "red")
+			src.audible_message("<span class='combat'><B>[src] buzzes oddly!</B></span>")
+			playsound(src.loc, "sparks", 50, 1, -1)
+			whitelist += blacklist
+			src.emagged = TRUE
+			return 1
+		return 0
+
+	demag(var/mob/user)
+		if (!src.emagged)
+			return 0
+		if (user)
+			user.show_text("You repair [src]'s material processor.", "blue")
+		src.emagged = FALSE
+		src.process_rate = min(src.process_rate, 3)
+		whitelist = initial(whitelist)
+		return 1
+
+	// attack by item places it in to disposal
+	attackby(obj/item/I, mob/user)
+		if(status & BROKEN)
+			return
+		if(iswrenchingtool(I) || istype(I,/obj/item/device/analyzer/atmospheric) || istype(I,/obj/item/card/emag))
+			..()
+			return
+		if(istype(I,/obj/item/electronics/scanner) || istype(I,/obj/item/deconstructor) || (istype(I,/obj/item/device/pda2)))
+			user.visible_message("<span class='alert'><B>[user] hits [src] with [I]!</B></span>")
+			return
```

Comment:
Sanity check on not calling `..()` here. I would think that we'd want to, but if intent is to not allow scanning/deconstructing I guess?

Reply:
Those are handled by `afterattack` so the context is the item doing the action, not the item receiving the action.